LEAF_Request_Portal/templates/view_reports.tpl

```diff
@@ -536,6 +580,17 @@ function editLabels() {
             	resIndicatorList[resSelectList[i]] = temp;
             }
         }
+        gridColorData = Object.assign({}, tempColorData);
+        if(gridColorData != {}) {
+            updateHeaderColors();
+            urlColorData = LZString.compressToBase64(JSON.stringify(gridColorData));
+            url = baseURL + '&v=' + version + '&query=' + encodeURIComponent(urlQuery) + '&indicators=' + encodeURIComponent(urlIndicators) + '&colors=' + encodeURIComponent(urlColorData);
+            if ($('#reportTitle').val() != '') {
+                url += '&title=' + encodeURIComponent(btoa($('#reportTitle').val()));
+            }
+            window.history.pushState('', '', url);
+        }
```

Comment:
Does this need an "else"?  What happens if gridColorData _is_ {}.  Even if nothing, would things look better/smoother if we had the option to define values outside of the code?

Reply:
Title and colors are situations where there might not be any info to add to the url (so if they are empty nothing).  There already is a keyup listener on the report title input field that changes the url so I needed to rebuild it here too (the Edit Labels save btn).  Open to ways to make it look cleaner though.